src/pages/list/search/articles/model.ts

```diff
@@ -0,0 +1,69 @@
+import { AnyAction, Reducer } from 'redux';
+
+import { EffectsCommandMap } from 'dva';
+import { ListItemDataType } from './data.d';
+import { queryFakeList } from './service';
+
+export interface StateType {
+  list: ListItemDataType[];
+}
+
+export type Effect = (
+  action: AnyAction,
+  effects: EffectsCommandMap & { select: <T>(func: (state: StateType) => T) => T },
+) => void;
+
+export interface ModelType {
+  namespace: string;
+  state: StateType;
+  effects: {
+    fetch: Effect;
+    appendFetch: Effect;
+  };
+  reducers: {
+    queryList: Reducer<StateType>;
+    appendList: Reducer<StateType>;
+  };
+}
+
+const Model: ModelType = {
+  namespace: 'listAndsearchAndarticles',
+
+  state: {
+    list: [],
+  },
+
+  effects: {
+    *fetch({ payload }, { call, put }) {
+      const response = yield call(queryFakeList, payload);
+      yield put({
+        type: 'queryList',
+        payload: Array.isArray(response) ? response : [],
+      });
+    },
+    *appendFetch({ payload }, { call, put }) {
+      const response = yield call(queryFakeList, payload);
+      yield put({
+        type: 'appendList',
+        payload: Array.isArray(response) ? response : [],
+      });
+    },
+  },
+
+  reducers: {
+    queryList(state, action) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected parameter: 'action' to have a typedef](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)